nautobot/extras/models/customfields.py

```diff
@@ -315,18 +312,16 @@ class CustomField(BaseModel, ChangeLoggedModel, NotesMixin):
         default=CustomFieldTypeChoices.TYPE_TEXT,
         help_text="The type of value(s) allowed for this field.",
     )
-    # 2.0 TODO: #824 remove `name` field as redundant, make `label` mandatory, populate `slug` from `label` field.
-    name = models.CharField(max_length=50, unique=True, help_text="Human-readable unique name of this field.")
     label = models.CharField(
         max_length=50,
-        blank=True,
-        help_text="Name of the field as displayed to users (if not provided, the field's name will be used.)",
+        help_text="Name of the field as displayed to users.",
+        blank=False,
```

Comment:
Should we think about making `label` be `unique=True` also? I know it'd be more impact in the data migration, not sure if worth the effort at this time.

Reply:
IMHO I think since `label` is user-facing and almost for display purposes only. (plus it can be modified anyway), it is probably not worth the effort to make it unique. But I could be swayed either way.

Reply:
Filter field label for CustomField models does derive from `cf.label`, so we would have duplicate filter field label if we allow duplicate `cf.label`. That again is a display issue.
![Screen Shot 2023-03-20 at 3 08 51 PM](https://user-images.githubusercontent.com/46973263/226441754-06dc87ce-be34-4802-9037-57d4400d5170.png)
The filter tags are constructed by` cf.key` tho
![Screen Shot 2023-03-20 at 3 09 07 PM](https://user-images.githubusercontent.com/46973263/226441771-143a18e2-12a6-42a3-af13-23ae3c16ce45.png)